contracts/zaps/steth/ZapSteth.sol

```diff
@@ -0,0 +1,190 @@
+// SPDX-License-Identifier: AGPL-3.0
+pragma solidity 0.6.12;
+pragma experimental ABIEncoderV2;
+
+import "../../../interfaces/curve/Curve.sol";
+import "../../../interfaces/lido/ISteth.sol";
+
+import "@openzeppelin/contracts/math/Math.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+
+import {
+    SafeERC20,
+    SafeMath,
+    IERC20,
+    Address
+} from "@openzeppelin/contracts/token/ERC20/SafeERC20.sol";
+import "../BaseZap.sol";
+
+contract ZapSteth is BaseZap {
+    using SafeERC20 for IERC20;
+    using Address for address;
+    using SafeMath for uint256;
+
+    ISteth public stETH =
+        ISteth(address(0xae7ab96520DE3A18E5e111B5EaAb095312D7fE84));
+
+    //slippage allowance is out of 10000. 20 is 2%
+    ICurveFi public StableSwapSTETH =
+        ICurveFi(address(0xDC24316b9AE028F1497c275EB9192a3Ea0f67022));
+
+    uint256 public constant DEFAULT_SLIPPAGE = 50;
+    bool private _noReentry = false;
+
+    constructor() public Ownable() {
+        want = IERC20(address(0x06325440D014e39736583c165C2963BA99fAf14E));
+        yVault = IYVault(address(0xdCD90C7f6324cfa40d7169ef80b12031770B4325));
+
+        want.safeApprove(address(yVault), uint256(-1));
+        want.safeApprove(address(StableSwapSTETH), uint256(-1));
+        stETH.approve(address(StableSwapSTETH), uint256(-1));
```

Comment:
maybe we can hardcode the list of allowed erc20s here or on the base Contract, wdyt?

Reply:
I think that's going to be really specific to every zap, look at the https://github.com/iearn-finance/zaps/pull/19 it has plenty of token approve.